File: 053_maximum_subarray/Solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        if (nums == null || nums.length == 0) return 0;

        int current = 0;
        int maxSum = Integer.MIN_VALUE;

        for (int num : nums) {
            current += num;
            maxSum = Math.max(current, maxSum);
            current = Math.max(0, current);
        }

        return maxSum;
    }

    public static void main(String[] args) {
        int[] nums = {-2, 1, -3, 4, -1, 2, 1, -5, 4};
        System.out.println(new Solution().maxSubArray(nums));
    }
}

File: 053_maximum_subarray/Solution.py
class Solution:
    def maxSubArray(self, nums):
        if not nums or len(nums) == 0: return 0

        current = 0
        max_sum = float('-inf')

        for num in nums:
            current += num
            max_sum = max(current, max_sum)
            current = max(current, 0)

        return max_sum


if __name__ == '__main__':
    print(Solution().maxSubArray([-2, 1, -3, 4, -1, 2, 1, -5, 4]))

